src/docfx/lib/JsonUtility.cs

```diff
@@ -482,6 +482,31 @@ private static void SetFieldWritable(MemberInfo member, JsonProperty prop)
             }
         }
 
+        private sealed class SerializerContractResolver : CamelCasePropertyNamesContractResolver
+        {
+            protected override JsonProperty CreateProperty(MemberInfo member, MemberSerialization memberSerialization)
+            {
+                var prop = base.CreateProperty(member, memberSerialization);
+                var contract = DefaultSerializer.ContractResolver.ResolveContract(prop.PropertyType);
+                if (contract is JsonArrayContract)
+                {
+                    prop.ShouldSerialize =
+                          value =>
+                          {
+                              var obj = JToken.FromObject(value);
+                              var array = (obj as JObject)?.GetValue(prop.PropertyName) as JArray;
+
+                              if (array?.Count == 0)
+                              {
+                                  return false;
+                              }
+                              return true;
+                          };
+                }
+                return prop;
+            }
+        }
+
         private sealed class JsonContractResolver : DefaultContractResolver
```

Comment:
This this also derive from `SerializerContractResolver`?

Reply:
intended to split these two, should deserializer be camel case?